### PR TITLE
fold ElmFormat.Cli{,Test} into ElmFormat.Integration

### PR DESCRIPTION
**Pull request by robx**
This moves the two test cases that require `ElmFormat.Cli` more-or-less faithfully into `Integration`, which runs against the proper `ElmFormat.main` implementation. With minimal extensions to `TestWorld`. To deal with those `ElmFormat.Cli` compile warnings. Joint work with @alicebob.

### CODE REVIEW

**Comment by robx** [comment]
(Not sure we want to keep the golden test call out to `diff`, used that for debugging, but it's sort of nice.)

**Comment by alicebob** [comment]
For #302 again!

**Comment by avh4** [comment]
This also relates to https://github.com/avh4/elm-format/issues/72

File: tests/Integration/CliTest.hs [hunk 1]
```diff
@@ -16,12 +18,27 @@ tests =
             world
                 |> run "elm-format" [ "--help" ]
                 |> expectExit 0
+        , world
+            |> run "elm-format-xxx" []
+            |> TestWorld.goldenStdout "usage instructions"
+                "tests/Integration/data/usage.stdout"
```
Comment (avh4) [inline tests/Integration/CliTest.hs]: I think you forgot to check this file in, or else forgot to change this path when you moved the file?

**Comment by avh4** [comment]
Looks like there's one mistake here (see comment above).  Otherwise, this looks great!

**Comment by robx** [comment]
Thanks for catching that! Indeed, I was moving files around later and tests were passing due to an untracked copy. Pushed a fix. Should I amend it into the buggy commit?

**Comment by alicebob** [comment]
But Travis passed anyway?

**Comment by avh4** [comment]
No need to amend.

Travis passed originally because apparently the behavior of golden is to create the file (and not error) if the file doesn't exist.

**Comment by avh4** [comment]
Thanks!

**Pull request merged**